Remove Microsoft.Diagnostics.Tracing.EventSource.Redist

noahfalk opened the pull request:
This package was used historically to distribute EventSource fixes out-of-band but
is no longer needed. This change removed the project and  the ES_BUILD_STANDALONE +
FEATURE_MANAGED_ETW_CHANNELS ifdefs which were only defined when the project was building.

Fixes #71231

noahfalk commented:
@brianrob @tarekgh - PTAL

tarekgh commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventProvider.cs:
```diff
@@ -467,7 +455,7 @@ unsafe void GetSessionInfo(SessionInfoCallback action, ref List<SessionInfo>? se
 
             // However the framework version of EventSource DOES have ES_SESSION_INFO defined and thus
```
>ES_SESSION_INFO [](http://example.com/codeflow?start=70&length=15)

need to update this comment?

tarekgh commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs:
```diff
@@ -21,26 +16,17 @@
-using System.Runtime.CompilerServices;
-namespace Microsoft.Diagnostics.Tracing
-#else
 using System.Runtime.CompilerServices;
```
>using System.Runtime.CompilerServices [](http://example.com/codeflow?start=0&length=37)

This should stay?

noahfalk replied:
That using statement was included once inside #if branch and a 2nd time in the #else branch. Eliminating the #if means it is now included once unconditionally. That looks correct to me, am I overlooking an issue?

tarekgh replied:
no, I am the one who was overlooking here :-)

tarekgh commented:
Looks the changes causing a compat issue, maybe we need to keep this API in the implementation. 

```
Build FAILED.

/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.22327.2/build/Microsoft.DotNet.ApiCompat.targets(52,5): error : Compat issues with assembly System.Diagnostics.Tracing: [/__w/1/s/src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.22327.2/build/Microsoft.DotNet.ApiCompat.targets(52,5): error : MembersMustExist : Member 'public System.Diagnostics.Tracing.EventChannel System.Diagnostics.Tracing.EventAttribute.Channel.get()' does not exist in the implementation but it does exist in the contract. [/__w/1/s/src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.22327.2/build/Microsoft.DotNet.ApiCompat.targets(52,5): error : MembersMustExist : Member 'public void System.Diagnostics.Tracing.EventAttribute.Channel.set(System.Diagnostics.Tracing.EventChannel)' does not exist in the implementation but it does exist in the contract. [/__w/1/s/src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.22327.2/build/Microsoft.DotNet.ApiCompat.targets(52,5): error : MembersMustExist : Member 'public System.Diagnostics.Tracing.EventChannel System.Diagnostics.Tracing.EventWrittenEventArgs.Channel.get()' does not exist in the implementation but it does exist in the contract. [/__w/1/s/src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]
/__w/1/s/.packages/microsoft.dotnet.apicompat/7.0.0-beta.22327.2/build/Microsoft.DotNet.ApiCompat.targets(66,5): error : ApiCompat failed for '/__w/1/s/artifacts/bin/System.Diagnostics.Tracing/Release/net7.0/System.Diagnostics.Tracing.dll' [/__w/1/s/src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj]
    0 Warning(s)
    5 Error(s)

```

tarekgh commented:
CC @ericstj for awareness.

stephentoub commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventProvider.cs:
```diff
@@ -147,7 +135,7 @@ internal unsafe void Register(EventSource eventSource)
             {
-#if TARGET_WINDOWS && !ES_BUILD_STANDALONE
+#if TARGET_WINDOWS
                 throw new ArgumentException(Interop.Kernel32.GetMessage(unchecked((int)status)));
```
Can this use the Marshal.GetPInvokeErrorMessage and not require ifdef'ing?

noahfalk replied:
I can't quickly guarantee it would have identical behavior and I am hoping to keep this change as a mechanical refactoring. GetPInvokeErrorMessage calls Interop.Sys.StrError(error) whereas currently the code invokes Convert.ToString(). I doubt EventRegister on non-Windows platforms was written to return an error code that strerror properly handles.

brianrob commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:
```diff
@@ -2338,16 +2255,12 @@ private bool IsEnabledByDefault(int eventNum, bool enable, EventLevel currentLev
             EventLevel eventLevel = (EventLevel)m_eventData[eventNum].Descriptor.Level;
             EventKeywords eventKeywords = unchecked((EventKeywords)((ulong)m_eventData[eventNum].Descriptor.Keywords & (~(SessionMask.All.ToEventKeywords()))));
 
-#if FEATURE_MANAGED_ETW_CHANNELS
```
FEATURE_MANAGED_ETW_CHANNELS is defined in coreclr/clr.featuredfines.props and and mono/System.Private.CoreLib/System.Private.CoreLib.csproj.  Are we sure that this isn't propagating?

brianrob replied:
Looks like it's also defined in [https://github.com/dotnet/runtime/blob/215b39abf947da7a40b0cb137eab4bceb24ad3e3/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs](https://github.com/dotnet/runtime/blob/215b39abf947da7a40b0cb137eab4bceb24ad3e3/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs).

noahfalk replied:
I think I was mistaken, I am planning to put it this ifdef back

ericstj commented:
> CC @ericstj for awareness.

I am OK with the premise of this PR, but it needs to be done in a way that doesn't break the package/API we still ship.

noahfalk commented:
> I am OK with the premise of this PR, but it needs to be done in a way that doesn't break the package/API we still ship.

For sure. I think it was a mistake on the FEATURE_MANAGED_ETW_CHANNELS which was defined elsewhere and the local builds I had done didn't have the right configurations to spot it.

ericstj commented:
You should be able to repro locally with a build of libs, then `dotnet build` of `src/libraries/System.Diagnostics.Tracing/src/System.Diagnostics.Tracing.csproj` which should build everything with minimal overhead.

noahfalk committed:
This package was used historically to distribute EventSource fixes out-of-band but

is no longer needed. This change removed the project and the ES_BUILD_STANDALONE
ifdef which was only defined when the project was building.

Fixes #71231

noahfalk committed:
More dead code elimination

noahfalk committed:
Feedback

noahfalk commented:
Build failure is known issue: https://github.com/dotnet/runtime/issues/71684

noahfalk commented:
I think all the feedback has been addressed, does this look good now? Thanks!

brianrob reviewed (approved):
LGTM!

noahfalk commented:
Thanks all! I'm merging on red but the issue shown by build analysis is a known one unrelated to this change: https://github.com/dotnet/runtime/issues/71684
